test/e2e/reaper_test.go

```diff
@@ -35,6 +39,36 @@ func createSingleReaper(t *testing.T, ctx context.Context, namespace string, f *
 	checkReaperReady(t, f, ctx, reaperKey)
 	checkReaperK8cStatusReady(t, f, ctx, kcKey, dcKey)
 
+	// check that the Reaper Vector container and config map exist
+	checkVectorAgentPresence(t, ctx, f, reaperKey, getPodTemplateSpecForReaper, reaper.VectorContainerName)
+	checkVectorAgentConfigMapPresence(t, ctx, f, dcKey, reaper.VectorAgentConfigMapName)
+
+	// check that if vector is disabled on reaper, the agent and configmap are deleted
+	t.Logf("check that if Reaper Vector is disabled, the agent and configmap are deleted")
+	err := f.Client.Get(ctx, kcKey, kc)
+	require.NoError(err, "failed to get K8ssandraCluster in namespace %s", namespace)
+	reaperVectorPatch := client.MergeFromWithOptions(kc.DeepCopy(), client.MergeFromWithOptimisticLock{})
+	kc.Spec.Reaper.Telemetry.Vector.Enabled = pointer.Bool(false)
+	err = f.Client.Patch(ctx, kc, reaperVectorPatch)
+	require.NoError(err, "failed to patch K8ssandraCluster in namespace %s", namespace)
+	checkReaperReady(t, f, ctx, reaperKey)
+	checkReaperK8cStatusReady(t, f, ctx, kcKey, dcKey)
+	checkVectorAgentDeleted(t, ctx, f, reaperKey, getPodTemplateSpecForReaper, reaper.VectorContainerName)
+	checkVectorConfigMapDeleted(t, ctx, f, dcKey, reaper.VectorAgentConfigMapName)
+
+	// check that if vector is re-enabled on reaper, the agent and configmap are re-added
+	t.Logf("check that if Reaper Vector is enabled, the agent and configmap are re-created")
+	err = f.Client.Get(ctx, kcKey, kc)
+	require.NoError(err, "failed to get K8ssandraCluster in namespace %s", namespace)
+	reaperVectorPatch = client.MergeFromWithOptions(kc.DeepCopy(), client.MergeFromWithOptimisticLock{})
+	kc.Spec.Reaper.Telemetry.Vector.Enabled = pointer.Bool(true)
+	err = f.Client.Patch(ctx, kc, reaperVectorPatch)
+	require.NoError(err, "failed to patch K8ssandraCluster in namespace %s", namespace)
+	checkReaperReady(t, f, ctx, reaperKey)
+	checkReaperK8cStatusReady(t, f, ctx, kcKey, dcKey)
+	checkVectorAgentPresence(t, ctx, f, reaperKey, getPodTemplateSpecForReaper, reaper.VectorContainerName)
```

Comment:
I know it's not part of this PR, but looking at the implementation of the function:
```go
func checkVectorAgentPresence(t *testing.T, ctx context.Context, f *framework.E2eFramework, podKey framework.ClusterKey, specFunction func(t *testing.T, ctx context.Context, f *framework.E2eFramework, dcKey framework.ClusterKey) *corev1.PodTemplateSpec, vectorContainerName string) {
	t.Logf("check that %s contains Vector Agent %s", podKey.Name, vectorContainerName)
	podTempSpec := specFunction(t, ctx, f, podKey)
	_, containerFound := cassandra.FindContainer(podTempSpec, vectorContainerName)
	require.True(t, containerFound, "cannot find Vector agent container in pod template spec")
}
```
It doesn't look like there's anything specific to Vector, it only checks the container by name. So maybe we shouldn't have "Vector" in the function name.

Reply:
I suppose we can. I wrote this one for the Stargate Vector work, along with the 2 functions that retrieve the PodTemplateSpec from the [CassDc](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/e2e/suite_test.go#L2049-L2054) and the Stargate [deployment](https://github.com/k8ssandra/k8ssandra-operator/blob/main/test/e2e/suite_test.go#L2056-L2061). But I guess it doesn't matter if the container it's looking for is a Vector container or something else.